src/d3a/d3a_core/sim_results/bills.py

```diff
@@ -142,142 +124,139 @@ def __init__(self, is_spot_market=True):
-            self.external_trades[area.name]['earned'] += \
-                (trade.offer.price - fee_price) / 100.
-            self.external_trades[area.name]['total_cost'] -= \
-                (trade.offer.price - fee_price) / 100.
-        self.external_trades[area.name]['total_energy'] -= trade.offer.energy
-        self.external_trades[area.name]['market_fee'] += fee_price / 100.
-
-    def _store_incoming_external_trade(self, trade, area):
-        fee_price = trade.fee_price / 100. if trade.fee_price is not None else 0.
-        self.external_trades[area.name]['bought'] += trade.offer.energy
+            self.external_trades[area_dict['name']]['earned'] += \
+                (trade['price'] - fee_price) / 100.
+            self.external_trades[area_dict['name']]['total_cost'] -= \
+                (trade['price'] - fee_price) / 100.
+        self.external_trades[area_dict['name']]['total_energy'] -= trade['energy']
+        self.external_trades[area_dict['name']]['market_fee'] += fee_price / 100.
+
+    def _store_incoming_external_trade(self, trade, area_dict):
+        trade_price = trade['price'] / 100.
+        fee_price = trade['fee_price'] / 100. if trade['fee_price'] is not None else 0.
+        self.external_trades[area_dict['name']]['bought'] += trade['energy']
         if ConstSettings.IAASettings.MARKET_TYPE == 1:
-            self.external_trades[area.name]['spent'] += \
-                trade.offer.price / 100.
-            self.external_trades[area.name]['total_cost'] += \
-                trade.offer.price / 100. + fee_price
+            self.external_trades[area_dict['name']]['spent'] += trade_price
+            self.external_trades[area_dict['name']]['total_cost'] += trade_price + fee_price
         else:
-            self.external_trades[area.name]['spent'] += \
-                trade.offer.price / 100. - fee_price
-            self.external_trades[area.name]['total_cost'] += \
-                trade.offer.price / 100.
-        self.external_trades[area.name]['total_energy'] += trade.offer.energy
+            self.external_trades[area_dict['name']]['spent'] += trade_price - fee_price
+            self.external_trades[area_dict['name']]['total_cost'] += trade_price
+        self.external_trades[area_dict['name']]['total_energy'] += trade['energy']
 
     @classmethod
-    def _default_area_dict(cls, area):
+    def _default_area_dict(cls, area_dict):
         return dict(bought=0.0, sold=0.0,
                     spent=0.0, earned=0.0,
                     total_energy=0.0, total_cost=0.0,
                     market_fee=0.0,
-                    type=area.display_type)
+                    type=area_dict['type'])
 
-    def _get_child_data(self, area):
-        if area.name not in self.bills_results:
-            self.bills_results[area.name] =  \
-                {child.name: self._default_area_dict(child)
-                    for child in area.children}
+    def _get_child_data(self, area_dict, area_core_stats):
```

Comment:
AFAIS, you dont need the `area_core_stats` in this function, right?

Reply:
Yes, removed!